src/core/java/app/com/linkedin/playparseq/j/stores/ParSeqTaskStoreImpl.java

```diff
@@ -33,30 +35,40 @@
   /**
    * The field ARGUMENTS_KEY is the default key of ParSeq Tasks.
    */
-  public final static String ARGUMENTS_KEY = "ParSeqTasks";
+  public final static TypedKey<Set<Task<?>>> ARGUMENTS_KEY = TypedKey.create("ParSeqTasks");
 
   /**
    * {@inheritDoc}
    */
   @Override
   public void put(final Http.Context context, final Task<?> task) {
-    get(context).add(task);
+    getOptional(context).map(tasks -> tasks.add(task));
   }
 
   /**
    * {@inheritDoc}
    */
-  @SuppressWarnings("unchecked")
   @Override
   public Set<Task<?>> get(final Http.Context context) {
```

Comment:
It's probably worth documenting that this will return an unmodifiable set and should be used for reading only, and that `getOptional` should be used for any modifications

Reply:
Updated!